x/profiles/types/models_chain_links.go

```diff
@@ -202,6 +204,51 @@ func (b Base58Address) VerifyPubKey(key cryptotypes.PubKey) (bool, error) {
 
 // --------------------------------------------------------------------------------------------------------------------
 
+var _ AddressData = &EthAddress{}
+
+// NewEthAddress returns a new EthAddress instance
+func NewEthAddress(value, prefix string) *EthAddress {
+	return &EthAddress{Value: value, Prefix: prefix}
+}
+
+func (e EthAddress) Validate() error {
+	if len(strings.TrimSpace(e.Value)) <= len(e.Prefix) {
+		return fmt.Errorf("address cannot be smaller than prefix")
+	}
+
+	prefix, addrWithoutPrefix := e.Value[:len(e.Prefix)], e.Value[len(e.Prefix):]
+	if prefix != e.Prefix {
+		return fmt.Errorf("prefix does not match")
+	}
+
+	if _, err := hex.DecodeString(addrWithoutPrefix); err != nil {
+		return fmt.Errorf("invalid Eth address")
+	}
+	return nil
+}
+
+// GetValue implements AddressData
+func (e EthAddress) GetValue() string {
+	return e.Value
+}
+
+// VerifyPubKey implements AddressData
+func (e EthAddress) VerifyPubKey(key cryptotypes.PubKey) (bool, error) {
+	addr := e.Value[len(e.Prefix):]
+	bz, err := hex.DecodeString(addr)
+	if err != nil {
+		return false, err
+	}
+	pub, err := btcec.ParsePubKey(key.Bytes(), btcec.S256())
+	if err != nil {
+		return false, err
+	}
+	uncompressedPub := pub.SerializeUncompressed()
+	return bytes.Equal(crypto.Keccak256(uncompressedPub[1:])[12:], bz), err
```

Comment:
Ethereum address is generated by uncompressed public key which is 65 bytes.

Reference: https://goethereumbook.org/wallet-generate/